rasterio/_base.pyx

```diff
@@ -831,6 +840,31 @@ cdef class DatasetBase(object):
 
         return GDALChecksumImage(band, xoff, yoff, width, height)
 
+    def get_gcps(self):
+        """Get GCPs and their associated CRS."""
+        cdef const char *wkt_b = GDALGetGCPProjection(self.handle())
+        if wkt_b == NULL:
+            raise ValueError("Unexpected NULL spatial reference")
+        wkt = wkt_b
+        crs = self._handle_crswkt(wkt)
+
+        cdef const GDAL_GCP *gcplist = NULL
+        gcplist = GDALGetGCPs(self.handle())
+        num_gcps = GDALGetGCPCount(self.handle())
+
+        return ([GroundControlPoint(col=gcplist[i].dfGCPPixel,
```

Comment:
Can we include the `GroundControlPoint.id` and `GroundControlPoint.info` as well?